src/commands/protocols/publish/receiver/v1.0.0/v1-0-0-handle-store-request-command.js

```diff
@@ -4,58 +4,113 @@ import {
-
-        const { assertionId: storeInitAssertionId } =
-            await this.operationIdService.getCachedOperationIdData(operationId);
-
-        if (storeInitAssertionId !== assertionId) {
-            throw Error(
-                `Store request assertion id ${assertionId} does not match store init assertion id ${storeInitAssertionId}`,
-            );
-        }
+        const {
+            blockchain,
+            keyword,
+            hashFunctionId,
+            contract,
+            tokenId,
+            operationId,
+            assertionId,
+            agreementId,
+            agreementData,
+        } = commandData;
 
         await this.operationIdService.updateOperationIdStatus(
             operationId,
             OPERATION_ID_STATUS.PUBLISH.VALIDATING_ASSERTION_REMOTE_START,
         );
-        const { blockchain, contract, tokenId } = this.ualService.resolveUAL(ual);
         const { assertion } = await this.operationIdService.getCachedOperationIdData(operationId);
         await this.operationService.validateAssertion(assertionId, blockchain, assertion);
 
         await this.operationIdService.updateOperationIdStatus(
             operationId,
             OPERATION_ID_STATUS.PUBLISH.VALIDATING_ASSERTION_REMOTE_END,
         );
+
         await this.operationIdService.updateOperationIdStatus(
             operationId,
             OPERATION_ID_STATUS.PUBLISH.PUBLISH_LOCAL_STORE_START,
         );
 
+        const ual = this.ualService.deriveUAL(blockchain, contract, tokenId);
+        const assetExists = await this.tripleStoreModuleManager.assetExists(
+            ual,
+            blockchain,
+            contract,
+            tokenId,
+        );
+
+        const agreementEndTime =
+            agreementData.startTime + agreementData.epochsNumber * agreementData.epochsLength;
+
         await this.operationService.localStoreAsset(
             assertionId,
             blockchain,
             contract,
             tokenId,
             operationId,
+            agreementData.startTime,
+            agreementEndTime,
+            keyword,
         );
+
         await this.operationIdService.updateOperationIdStatus(
             operationId,
             OPERATION_ID_STATUS.PUBLISH.PUBLISH_LOCAL_STORE_END,
         );
 
+        await this.repositoryModuleManager.updateOperationAgreementStatus(
+            operationId,
+            agreementId,
+            AGREEMENT_STATUS.ACTIVE,
+        );
+
+        if (!assetExists) {
+            this.logger.trace(
+                `Asset with ${ual} not previously present in triple store. Scheduling epoch check command.`,
+            );
+            await this.commandExecutor.add({
+                name: 'epochCheckCommand',
+                sequence: [],
+                delay: 0,
+                data: {
+                    blockchain,
+                    contract,
+                    tokenId,
+                    keyword,
+                    epoch: 0,
```

Comment:
Why are we setting epoch to 0 for received store request?

Reply:
Because when the node receives the data the epoch is 0